### PR TITLE
feat: support for tweakpane v4

### PR DESCRIPTION
**Pull request by kitschpatrol**
Hi, similar to [this PR](https://github.com/0b5vr/tweakpane-plugin-rotation/pull/2) on your other plugin, I've updated the Profiler for compatibility with Tweakpane 4.x.

More involved changes were required than for the Rotation plugin, due to more significant shifts in Tweakpane 4's  blade implementation.

Thanks again for making these useful tools open-source.

### CODE REVIEW

**Comment by 0b5vr** [comment]
I will review this as well, thanks a lot! + I'm sorry to be unaware for a while 🙇

File: src/ProfilerApi.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
 import { BladeApi } from '@tweakpane/core';
-import type { LabelController } from '@tweakpane/core';
-import type { ProfilerBladeController } from './ProfilerBladeController';
-import type { ProfilerBladeMeasureHandler } from './ProfilerBladeMeasureHandler';
+import type { ProfilerBladeController } from './ProfilerBladeController.js';
+import type { ProfilerBladeMeasureHandler } from './ProfilerBladeMeasureHandler.js';
 
-export class ProfilerBladeApi extends BladeApi<LabelController<ProfilerBladeController>> {
+export class ProfilerBladeApi extends BladeApi<ProfilerBladeController> {
```
Comment (0b5vr) [inline src/ProfilerApi.ts]: Feels like `ProfilerBladeApi` is more appropriate if we follow the name of the base class. Why was the file name changed?

File: src/ProfilerController.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+interface CalcCache {
+  meanCalc: HistoryMeanCalculator;
+  medianCalc: HistoryPercentileCalculator;
+  latest: number;
+  childrenCacheMap: ConsecutiveCacheMap<string, CalcCache>;
+  childrenPromiseDelta: Promise<number>[];
+}
+
+
+export class ProfilerController implements Controller<ProfilerView> {
```
Comment (0b5vr) [inline src/ProfilerController.ts:24]: huhh, now `BladeController` and `Controller` are different, I got it. Or is this a concept that the v3 already had?

Comment (0b5vr) [reply src/ProfilerController.ts:24]: Ahh, I might've simply not been aware of `BladeController` at the moment I implement this. Looks like a very nice improvement.

File: src/ProfilerBladeController.ts [hunk 3]
```diff
@@ -1,136 +1,26 @@
...
-  public bufferSize: number;
-  public measureHandler: ProfilerBladeMeasureHandler;
-  public readonly view: ProfilerBladeView;
-  public readonly viewProps: ViewProps;
-  private ticker_: Ticker;
-
-  private rootCalcCacheStack_: CalcCache[];
+export class ProfilerBladeController extends BladeController<LabelView> {
+  public readonly labelController: LabelController<ProfilerController>;
+  public readonly valueController: ProfilerController;
```
Comment (0b5vr) [inline src/ProfilerBladeController.ts:7]: It's called `valueController` since it is `valueController` even in `LabelController`. ✅

**Review by 0b5vr — CHANGES_REQUESTED** [review]
The filename `ProfilerApi.ts` is the only issue I want to check before merge!

File: src/ProfilerApi.ts (hunk 1 above)
Comment (kitschpatrol) [reply src/ProfilerApi.ts]: Oops yes `ProfilerBladeApi.ts` makes more sense.

**Review by 0b5vr — APPROVED** [review]
Thanks a lot!

**Pull request merged**